### ISSUE TITLE
Reproducible (minimal ?) need for two compileAttributes() runs

### ISSUE THREAD

**Issue by eddelbuettel** [RcppCore/Rcpp#934]
This comes courtesy of a colleague.  If one does

1. Create 'New package with Rcpp' followed by Build->Install+Restart -- OK
2. Add new C++ file using the examples, saved in `src/` as _.e.g._ `src/doubleMe.src, Install+Restart -- OK
3. Now delete the initial src/rcpp_hello_world.cpp, Install+Restart --> Link error

Two runs of `compileAttributes()` fix it, but the IDE does not suggest it.  Can we think of a heuristic to figure out when to run it twice?

**Comment by coatless** [comment]
@kevinushey already produced a reprex c.f. https://github.com/RcppCore/Rcpp/issues/733#issuecomment-314505646

**Comment by eddelbuettel** [comment]
Thanks for that -- I once knew that too.

**Comment by coatless** [comment]
tl;dr of the story: the bug effects ~1% < of the developer base and is quickly solved by re-running hence there never was a push to fix it.

**Comment by eddelbuettel** [comment]
I does frustrate users.  So I am wondering if we should just run the (still very fast) `compileAttributes()` twice.

**Comment by eddelbuettel** [comment]
Closed via #939
